### PR TITLE
MarathonStore: Log deserialization errors

### PR DESCRIPTION
**Pull request by kolloch**
I guess that we at least want to know about serialization errors, correct?

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/state/MarathonStore.scala [hunk 1]
```diff
@@ -113,7 +113,9 @@ class MarathonStore[S <: MarathonState[_, S]](
       Some(newState().mergeFromProto(bytes))
     }
     catch {
-      case e: InvalidProtocolBufferException => None
+      case e: InvalidProtocolBufferException =>
```
Comment (drexin) [inline src/main/scala/mesosphere/marathon/state/MarathonStore.scala]: Would it make sense to log the key for that state as well? I know it's not accessible from here, but maybe we should refactor this a bit.

Comment (kolloch) [inline src/main/scala/mesosphere/marathon/state/MarathonStore.scala]: Good point. thanks!

Comment (kolloch) [inline src/main/scala/mesosphere/marathon/state/MarathonStore.scala]: Ok, passing stateFromBytes the key for debugging is technically simple. From talking to you, I understood that you do not like that. I can follow you, even if I do not agree.

The alternative is to perform the error handling in the calling methods: modify and fetch.

Right now, they have really strange error handling semantics. If a key does not exist, an exception is thrown. If the data retrieved for the key cannot be read, the error is ignored and either None or a default empty object is returned.

I will change that but that of course will pretty much extend the scope of this patch.

**Comment by kolloch** [comment]
Ok, now I moved error reporting one level up and I throw StorageExceptions.

Maybe, I should stick to logging only. What do you think?

I had an intermediary version where I just passed the `key` name to `stateFromBytes` for the error log statements. It was pretty short an unintrusive but we had a discussion about that...

**Comment by drexin** [comment]
@kolloch Could you please rebase and fix this PR

**Comment by kolloch** [comment]
@drexin Yes :)

**Comment by drexin** [comment]
:+1:

**Pull request merged**